optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/BavetConstraintSession.java

```diff
@@ -88,6 +89,23 @@ public BavetConstraintSession(boolean constraintMatchEnabled, ScoreDefinition<Sc
         return nodeIndexedNodeMap;
     }
 
+    private static void refreshTuple(BavetAbstractTuple tuple) {
+        ((BavetAbstractNode) tuple.getNode()).refresh(tuple);
+        switch (tuple.getState()) {
+            case CREATING:
+            case UPDATING:
+                tuple.setState(BavetTupleState.OK);
+                return;
+            case DYING:
+            case ABORTING:
+                tuple.setState(BavetTupleState.DEAD);
+                return;
+            case DEAD:
+                throw new IllegalStateException(
+                        "The tuple (" + tuple + ") is already in the dead state (" + tuple.getState() + ").");
```

Comment:
Let's a default case that throws an exception?

Reply:
Also, this exception can start with "Impossible state:"
```suggestion
                        "Impossible state: The tuple (" + tuple + ") is already in the dead state (" + tuple.getState() + ").");
```

Reply:
Do we also want to print out the node toString in the exception message? Might be useful if someone ever sees it on productions and asks us what could be the cause.